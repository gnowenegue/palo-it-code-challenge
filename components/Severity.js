import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

const Severity = props => {
  const { id, emoticon, setSeverity } = props;
  useEffect(() => {
    document.querySelectorAll('input[type="radio"]').forEach(radioBtn => {
      radioBtn.addEventListener('change', e => {
        setSeverity(parseInt(e.target.value, 10));

        document
          .querySelector('#severity.modal .modal-footer a')
          .classList.remove('disabled');
      });
    });
  });

  return (
    <div className={`col s2 center-align ${id === 1 ? 'offset-s1' : ''}`}>
      <input value={id} id={`severity-${id}`} name="severity" type="radio" />
      <label htmlFor={`severity-${id}`} className="blue-text text-darken-4">
        {emoticon}
        <br />
        {id}
      </label>
      <style jsx>
        {`
          input[type='radio'] {
            display: none;
          }
          label {
            cursor: pointer;
          }
          input[type='radio']:checked + label {
            font-size: 3rem;
            transition: font-size 0.3s;
          }
          input[type='radio'] + label {
            font-size: 2rem;
            transition: font-size 0.3s;
          }
        `}
      </style>
    </div>
  );
};

Severity.propTypes = {
  id: PropTypes.number.isRequired,
  emoticon: PropTypes.string.isRequired,
  setSeverity: PropTypes.func.isRequired,
};

export default Severity;
