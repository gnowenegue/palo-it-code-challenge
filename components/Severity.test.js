import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow, render } from 'enzyme';

// import AppContext from '../contexts/AppContext';
import Severity from './Severity';

describe('<Severity />', () => {
  test('Snapshot test', () => {
    const render = renderer.create(
      <Severity id={1} emoticon="emoticon" setSeverity={() => {}} />
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
