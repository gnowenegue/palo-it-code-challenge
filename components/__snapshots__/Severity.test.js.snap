// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Severity /> Snapshot test 1`] = `
<div
  className="jsx-3703842634 col s2 center-align offset-s1"
>
  <input
    className="jsx-3703842634"
    id="severity-1"
    name="severity"
    type="radio"
    value={1}
  />
  <label
    className="jsx-3703842634 blue-text text-darken-4"
    htmlFor="severity-1"
  >
    emoticon
    <br
      className="jsx-3703842634"
    />
    1
  </label>
</div>
`;
